### PR TITLE
Modular PR: yield messages and flatten main iteration

### PR DESCRIPTION
**Pull request by CodyCBakerPhD**
## Motivation

What was the reasoning behind this change? Please explain the changes briefly.

Also seeing what errors show up in CI - ~~getting some odd errors on local device from the new try/except structure.~~ (something wasn't nested properly)

## Checklist

- [x] Have you ensured the PR description clearly describes the problem and solutions?
- [x] Have you checked to ensure that there aren't other open or previously closed [Pull Requests](https://github.com/neurodatawithoutborders/nwbinspector/pulls) for the same change?
- [x] Is your code contribution compliant with `black` format? If not, simply `pip install black` and run `black .` inside your fork.
- [x] If this PR fixes an issue, is the first line of the PR description `fix #XX` where `XX` is the issue number?

### CODE REVIEW

**Comment by CodyCBakerPhD** [comment]
Need to...
- [x] debug tests
- [x] add explicit tests for the iteration feature
- [x] ~~improve confige_checks testing~~ moved to separate PR

File: nwbinspector/nwbinspector.py [hunk 1]
```diff
@@ -36,6 +36,68 @@ def default(self, o):
...
+            checks_out.append(check)
+    else:
+        checks_out = checks
+
+    if select:
+        checks_out = [x for x in checks_out if x.__name__ in select]
+    elif ignore:
+        checks_out = [x for x in checks_out if x.__name__ not in ignore]
+    if importance_threshold:
+        checks_out = [x for x in checks_out if x.importance.value >= importance_threshold.value]
```
Comment (CodyCBakerPhD) [inline nwbinspector/nwbinspector.py:103]: Centralized these lines across all the calling functions so they only get called once, prior to any iteration.

File: nwbinspector/nwbinspector.py [hunk 2]
```diff
@@ -184,52 +231,63 @@ def inspect_nwb(
...
-        message.importance = Importance.ERROR
-        message.check_function_name = f"{type(ex)}: {str(ex)}"
-        message.file = file_name
-        messages.append(message)
-    return messages
+        except Exception as ex:
+            yield InspectorMessage(
+                message=traceback.format_exc(),
+                importance=Importance.ERROR,
+                check_function_name=f"{type(ex)}: {str(ex)}",
```
Comment (CodyCBakerPhD) [inline nwbinspector/nwbinspector.py:257]: This type of change was needed when using the new organization method with one of the levels being `check_function_name`.

**Comment by bendichter** [comment]
@CodyCBakerPhD ready for review?

**Comment by CodyCBakerPhD** [comment]
I'm discovering more and more bugs with the config functionality, which wasn't the main intention of this PR so I'll be splintering those changes into a separate PR so this one can focus exclusively on the yielding and main iteration logic.

ETA ~~next couple hours~~ within the hour

**Comment by CodyCBakerPhD** [comment]
@bendichter Ready for review; this PR upgrades the core `inspect_*` functions to always `yield`, updates the tests to demonstrate how to collapse the iteration of those functions to match output prior to this change.

Also demonstrates how to perform manual iteration using `next` (if the user of the library simply wants to know if there is at least one violation present) and assures that doing so on a violation-less NWBFile results in nothing more than a `StopIteration`.

**Review by bendichter — APPROVED** [review]

**Pull request merged**